src/test/lib/storage/index/adaptive_radix_tree/adaptive_radix_tree_index_test.cpp

```diff
@@ -219,24 +234,65 @@ TEST_F(AdaptiveRadixTreeIndexTest, SimpleTest) {
 * The following two cases try to test two rather extreme situations that both
 * test the node overflow handling of the ART implementation:
 *   - sparse vector: wide range of values (between 1 and MAX_INT) with large gaps
-*   - dense vector: expenential distribution, rounded to integer values
+*   - dense vector: exponential distribution, rounded to integer values
 **/
-TEST_F(AdaptiveRadixTreeIndexTest, SparseVectorOfRandomInts) {
-  size_t test_size = 1'000;
-  std::uniform_int_distribution<int32_t> uni(1, std::numeric_limits<int32_t>::max() - 1);
+TEST_F(AdaptiveRadixTreeIndexTest, SparseVectorOfInts) {
+  const auto test_size = 1'000u;
+  const auto cluster_number = 10u;  // 10 clusters with 9 gaps distributed over the range of [0, int32_max]
+  const auto cluster_size = test_size / cluster_number;
+  const auto gap_size = (std::numeric_limits<int32_t>::max() - cluster_number * cluster_size) / (cluster_number - 1);
+
+  std::vector<std::optional<int32_t>> values{};
+  values.reserve(test_size);
+
+  std::array<std::array<int32_t, cluster_size>, cluster_number> clustered_values{};
+
+  // Create clusters of values within the range [0, int32_t::max]. Each cluster has a start value and subsequent values
```

Comment:
Could you describe the rationale for creating these clusters?

Reply:
Creating these clusters is an approach to meet the requirement of large gaps between values in particular. But since std::uniform_int_distribution was used previously, we can also use the value pool approach that is used above. That would be less complex.

Reply:
Not saying this needs to change in any way. Just add the comment in the code instead of here.